source/material_model/rheology/elasticity.cc

```diff
@@ -273,78 +316,168 @@ namespace aspect
-                // properties (viscoelastic viscosity, shear modulus), elastic time step size, strain rate,
-                // vorticity, prior (inherited) viscoelastic stresses and viscosity of the elastic damper.
-                // In the absence of the elastic damper, the expression for "stress_new" is identical
-                // to the one found in Moresi et al. (2003, J. Comp. Phys., equation 29).
-                const double damped_elastic_viscosity = calculate_elastic_viscosity(average_elastic_shear_moduli[i]);
+                const Tensor<2,dim> rotation = 0.5 * ( velocity_gradients[i] - transpose(velocity_gradients[i]) );
 
                 // stress_0 is the combination of the elastic stress tensor stored at the end of the last time step and the change in that stress generated by local rotation
-                const SymmetricTensor<2,dim> stress_0 = (stress_old + dte * ( symmetrize(rotation * Tensor<2,dim>(stress_old) ) - symmetrize(Tensor<2,dim>(stress_old) * rotation) ) );
-
-                // stress_creep is the stress experienced by the viscous and elastic components.
-                const SymmetricTensor<2,dim> stress_creep = 2. * average_viscoelastic_viscosity * ( deviator(in.strain_rate[i]) + stress_0 / (2. * damped_elastic_viscosity ) );
-
-                // stress_new is the (new) stored elastic stress
-                SymmetricTensor<2,dim> stress_new = stress_creep * (1. - (elastic_damper_viscosity / damped_elastic_viscosity)) + elastic_damper_viscosity * stress_0 / damped_elastic_viscosity;
+                const SymmetricTensor<2,dim> stress_0 = (stress_t[i] +
+                                                         dte * ( symmetrize(rotation * Tensor<2,dim>(stress_t[i]) ) - symmetrize(Tensor<2,dim>(stress_t[i]) * rotation) ) );
 
                 // Stress averaging scheme to account for difference between fixed elastic time step
-                // and numerical time step (see equation 32 in Moresi et al., 2003, J. Comp. Phys.)
-                if (use_stress_averaging == true)
-                  {
-                    stress_new = ( ( 1. - ( dt / dte ) ) * stress_old ) + ( ( dt / dte ) * stress_new ) ;
-                  }
+                // and numerical time step (see equation 32 in Moresi et al., 2003, J. Comp. Phys.).
+                //
+                // Note that Moresi et al. incorrectly use the symbol $\tau_t$ for the 'newly calculated
+                // stress tensor' in the sentence above Eq. 32.
+                // Also note that in the Supplement of Sandiford et al. 2021 in Eq. (9) the new and
+                // old stress have been incorrectly switched. In Eq. (8) of the same supplement,
+                // the advected stress history tensor is used, but this tensor is only advected during
+                // the advection solve for which we are computing the reaction terms here (in case of fields).
+                // TODO Change: delete these lines. Move to somewhere else?
```

Comment:
I think these lines should be removed from the code, but the key points added into the ASPECT manual when we put in a section for the VEP rheology.

Reply:
Can you explain again why these lines are not needed anymore? If they are unnecessary or wrong, they should be deleted (no need to keep them). If they are a feature that is unsupported with the new way we do elastic stressses, we could keep a note about this (which feature, which PR it was removed), but still remove the code to avoid outdated code and confusion (we can always get it back from the git history).